### ISSUE TITLE
[Proposal] Option to switch table naming to camel case

### ISSUE THREAD

**Issue by Modelizer** [laravel/ideas#1133]
Wouldn't be nice to have an option in `config/database.php` to switch table naming convention to camel or underscore case? This would be a good feature to those who are using camel case because they might migrated from different framework or years back their database convention was camel case.

**Comment by tomschlick** [comment]
You can use the `$table` property on each model.

https://laravel.com/docs/5.6/eloquent#defining-models

**Comment by Modelizer** [comment]
I know that but wouldn't be nice to just configure it at one place and not repeating yourself in every model?

**Comment by BrandonShar** [comment]
You could just use a base model class that extends Eloquent and override `getTable` on it.

**Comment by Patryk27** [comment]
@BrandonShar: Yes, one could do it - but it doesn't answer why shouldn't it be configurable from the framework. You could as well say `why have database configurable from the .env file, while you can override the database factory`.

@Modelizer: it's a hands up for me - this case is not common, but when it happens, it would be nice to have it configurable.

**Comment by BrandonShar** [comment]
Except your example isn't the same situation at all. `env` and `config` are meant to handle values that can change based on the environment. Nearly every application is going to have different database connections/credentials based on the environment (most people don't use their production database in development, for example). Because these values change depending on the location of the app, it doesn't make sense to have them as part of the application logic.

Your table naming convention, on the other hand, is extremely unlikely to change between production and development, so it doesn't need to be an `env` variable, it can just be part of your application logic.

**Comment by BrandonShar** [comment]
Those both deal with concepts that are fairly deep in the framework and aren't easily modified in user-land. Determining the name of the underlying table is something that happens in your model that you created, not in framework level functionality.

**Comment by Patryk27** [comment]
@BrandonShar: Ok, agreed.

**Comment by Modelizer** [comment]
@BrandonShar creating an abstract model for just single functionality sound inappropriate for me. This is just a configuration nice to have.
